packages/cli/test/unit/ConcurrencyQueue.test.ts

```diff
@@ -0,0 +1,58 @@
+import { ConcurrencyQueue } from '@/ConcurrencyQueue';
+
+beforeAll(() => {
+	jest.useFakeTimers();
+});
+
+afterAll(() => {
+	jest.useRealTimers();
+});
+
+describe('ConcurrencyQueue', () => {
+	test('should limit concurrency', async () => {
```

Comment:
We're testing state at various Ts but we're not testing that the concurrency limit is being enforced?